modules/aws/src/smithy4s/query/FormData.scala

```diff
@@ -0,0 +1,55 @@
+package smithy4s.aws.query
+
+final case class FormData(
+    values: Map[List[String], String],
+    primitive: Option[String]
+) {
+  lazy val render: Option[String] =
+    primitive.orElse(
+      if (values.isEmpty) None
+      else
+        Some(
+          values
+            .map { case (segments, value) =>
+              segments.reverse.mkString(".") + "=" + value
```

Comment:
In theory you should never have to reverse, as you're always gonna be computing the fields first and prepending the name of the structure field or the union alternative

Reply:
Also, you're gonna have to URL-encode both the key and value here. You can find an encoder in `smithy4s.internals.URIEncoderDecoder`, which is package private, so you'll have to write the package of this file as : 

``` 
package smithy4s
package aws.query
```

Reply:
Done